src/Polynomials.jl

```diff
@@ -363,6 +365,7 @@ p(4)   # 57 = 1 + 2*4 + 3*4^2
 """
 function polyval{T,S}(p::Poly{T}, x::S)
     R = promote_type(T,S)
+    T<:AbstractFloat && any(isnan(p.a)) && convert(R, NaN)
```

Comment:
I see the appeal, but I just don't like it. People who want to use the package for different coefficient rings (such as Mod2) will find it confusing what all needs to be implemented. This puts a burden on people unfamiliar with the package. The runtime check is the only cost of doing it in this package. Alternatively, we can avoid overloading `isnan`, as polys aren't exactly "floating point numbers" and define our own `hasnan`, which is somewhat more descriptive. Maybe:

`hasnan{T <: AbsractFloat}(p::Poly{T}) = any(isnan(p.a))`
`hasnan{T <: AbsractFloat}(p::Poly{Complex{T}}) = any(isnan(p.a))`
`hasnan(p::Poly{T}) = false`

Then  we can check as `hasnan(p) && convert(R, NaN)`

I did something like this in `show.jl` with `hasneg` to see if a check that a coefficient is negative is reasonable for the underlying set of coefficients.